core/src/main/kotlin/plugins/OpenApi.kt

```diff
@@ -150,3 +159,163 @@ fun Application.configureOpenApi() {
         }
     }
 }
+
+private fun Bundle<SwaggerSchema>.withCustomTitle(): Bundle<SwaggerSchema> {
+    return CustomTitleStep().process(this)
+}
+
+/**
+ * An adapted version of [io.github.smiley4.schemakenerator.swagger.steps.SwaggerSchemaAutoTitleStep] that generates
+ * valid titles for the generated schemas. For example, the type `GenericType<Type1, Type2>` is transformed to
+ * `GenericType_Type1-Type2` instead of `GenericType<Type1,Type2>`.
+ *
+ * According to the [specification](https://swagger.io/specification/#components-object), schema keys must match the
+ * regular expression `^[a-zA-Z0-9\.\-_]+$`.
+ */
+private class CustomTitleStep {
+    fun process(bundle: Bundle<SwaggerSchema>): Bundle<SwaggerSchema> {
+        val typeDataMap = bundle.buildTypeDataMap()
+        return bundle.also { schema ->
+            process(schema.data, typeDataMap)
+            schema.supporting.forEach { process(it, typeDataMap) }
+        }
+    }
+
+    private fun process(schema: SwaggerSchema, typeDataMap: Map<TypeId, BaseTypeData>) {
+        if (schema.swagger.title == null) {
+            schema.swagger.title = determineTitle(schema.typeData, typeDataMap)
+        }
+    }
+
+    private fun determineTitle(typeData: BaseTypeData, typeDataMap: Map<TypeId, BaseTypeData>): String {
+        return typeData.simpleName.let {
+            if (typeData.typeParameters.isNotEmpty()) {
+                val paramString = typeData.typeParameters
+                    .map { (_, param) -> determineTitle(typeDataMap[param.type]!!, typeDataMap) }
+                    .joinToString("-")
+                "${it}_$paramString"
+            } else {
+                it
+            }
+        }.let {
+            it + (typeData.id.additionalId?.let { a -> "#$a" } ?: "")
+        }
+    }
+}
+
+private fun Bundle<SwaggerSchema>.customCompileReferencingRoot(): CompiledSwaggerSchema {
+    return CustomCompileReferenceRootStep().compile(this)
```

Comment:
Maybe turn into an expression?

Reply:
I didn't want to optimize the copied code, in case we have to integrate any upstream changes. But I also hope that the upstream issue will get fixed an we can delete it again.

Reply:
Sorry, I missed that this *all* was copied code. We can keep it as-is then IMO, in the the hope to soon remove it again.

Reply:
Should we then temporarily add "Copyright 2024 SMILEY4" to the header of this file?

Reply:
Yes, done.

Reply:
Of course, this now fails the check for the expected license header 🙄

Reply:
I have changed the rule to allow that additional copyright statement and added a note to the commit message.